### PR TITLE
atuin: 18.2.0 -> 18.3.0

### PR DESCRIPTION
**Pull request by 0x4A6F**
## Description of changes

Upgrate to [18.3.0](https://github.com/atuinsh/atuin/releases/tag/v18.3.0) including [these] (https://github.com/atuinsh/atuin/compare/v18.2.0...v18.3.0) changes.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by msanft — COMMENTED** [review]
I think this might need `protoc` now.

**Comment by 0x4A6F** [comment]
> I think this might need `protoc` now.

Yeah, was interrupted by work during my changes.

**Comment by msanft** [comment]
Result of `nixpkgs-review pr 319039` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>atuin</li>
  </ul>
</details>

**Comment by 0x4A6F** [comment]
> Result of `nixpkgs-review pr 319039` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
> 1 package failed to build:

Strange, on `b79dc36` I get:
```
0x4A6F@darwin01 ~/nixpkgs (git)-[master-atuin] % nix-build -A pkgs.atuin
/nix/store/yd43rcpgx11nd6aqs81ykzm83xa4phk8-atuin-18.3.0
```

**Comment by msanft** [comment]
> > Result of `nixpkgs-review pr 319039` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
> > 1 package failed to build:
> 
> Strange, on `b79dc36` I get:
> 
> ```
> 0x4A6F@darwin01 ~/nixpkgs (git)-[master-atuin] % nix-build -A pkgs.atuin
> /nix/store/yd43rcpgx11nd6aqs81ykzm83xa4phk8-atuin-18.3.0
> ```

Yeah, I think my remote builder on Darwin might be misconfigured. Gimme a sec

**Comment by msanft** [comment]
Result of `nixpkgs-review pr 319039` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>atuin</li>
  </ul>
</details>

**Review by msanft — APPROVED** [review]
LGTM, thanks!

File: pkgs/by-name/at/atuin/package.nix [hunk 1]
```diff
@@ -4,43 +4,51 @@
...
+    protobuf
+  ] ++ lib.optionals stdenv.isDarwin [
     libiconv
     darwin.apple_sdk_11_0.frameworks.AppKit
     darwin.apple_sdk_11_0.frameworks.Security
     darwin.apple_sdk_11_0.frameworks.SystemConfiguration
   ];
 
+  preBuild = ''
+    export PROTOC=${protobuf}/bin/protoc
```
Comment (SuperSandro2000) [inline pkgs/by-name/at/atuin/package.nix:48]: This should probably be buildPackages.protobuf or cross won't work

**Pull request merged**